src/main/java/seedu/duke/parser/TaskManager.java

```diff
@@ -0,0 +1,28 @@
+package seedu.duke.parser;
+
+import seedu.duke.command.ExitCommand;
+import seedu.duke.exception.GetJackDException;
+
+/**
+ * To execute task actions based on keywords used in user command
+ */
+public class TaskManager {
+    /**
+     * Processes the extracted user command
+     *
+     * @param command is the extracted first word of the user input
+     * @param input is the command given by the user
+     * @throws GetJackDException if invalid command is given
+     */
+    public static void parseUserCommand(String command, String input) throws GetJackDException {
+        switch(command) {
+        case "bye":
+            // Changes the loop condition to true to exit from the program
+            ExitCommand.isExit();
+            break;
+        default:
+            throw new GetJackDException("I'm sorry, I do not know what you are trying to say: " +
+                    "Please refer to the User Guide");
+        }
+    }
+}
```

Comment:
Perhaps this class could be merged with Parser since for now the only task it does is "parseUserCommand". Please ignore this suggestion if this class has a specialised function later in the future.

Or maybe can simply rename the method to executeUserCommand if that is what its meant to do

Reply:
We will be calling `executeUserCommand`, present in the `Command` classes, from this method. This is done to optimize the use OOP concept.

Reply:
In that case, maybe we could return a `Command` object in this method and execute it in the `Duke.run()` method?
Since it might be confusing for someone to see the command getting executed inside a "parse" function. Or we could rename it to `parseAndExecuteCommand()` (although that does make it a bit too long) so that people know exactly what the method does. Could also add it under the documentation but I feel changing its behaviour or name makes it easier to understand.

Reply:
Oh, I understand now 😄. Yup, both your ideas are good! I had the first one in mind but like you mentioned, the code quality can be done after v1.0

Reply:
yeah I agree with Ishaan on this one